plugins/colorbutton/plugin.js

```diff
@@ -499,26 +512,34 @@
 			 * * HSL/HSLA colors (e.g. `hsl( 100, 50%, 20%)` or `hsla( 100, 50%, 20%, 10%)`).
 			 *
 			 * @private
-			 * @param {String} color
+			 * @param {String/CKEDITOR.tools.color} color
 			 * @returns {String} Returns color in hex format, but without the hash at the beginning, e.g. `ff0000` for red.
+			 * In case of transparent colors, it returns `'transparent'` string.
 			 */
 			normalizeColor: function( color ) {
-				var alphaRegex = /^(rgb|hsl)a\(/g,
-					transparentRegex = /^rgba\((\s*0\s*,?){4}\)$/g,
-					isAlphaColor = alphaRegex.test( color ),
-					// Browsers tend to represent transparent color as rgba(0, 0, 0, 0), so we need to filter out this value.
-					isTransparentColor = transparentRegex.test( color ),
-					colorInstance;
-
-				// For colors with alpha channel we need to use CKEDITOR.tools.color normalization (#4351).
-				if ( isAlphaColor && !isTransparentColor ) {
-					colorInstance = new CKEDITOR.tools.color( color );
-
-					return CKEDITOR.tools.normalizeHex( colorInstance.getHex() || '' ).replace( /#/g, '' );
+				var colorInstance = ( color instanceof CKEDITOR.tools.color ) ? color : new CKEDITOR.tools.color( color, '' ),
+					isTransparent = Number( colorInstance._.alpha ) === 0;
+
+				if ( isTransparent ) {
+					return 'transparent';
 				}
 
 				// Replace 3-character hexadecimal notation with a 6-character hexadecimal notation (#1008).
-				return CKEDITOR.tools.normalizeHex( '#' + CKEDITOR.tools.convertRgbToHex( color || '' ) ).replace( /#/g, '' );
+				// It also covers other cases, like colors with alpha channel (#4351).
+				return colorInstance.getHex() || '';
+			},
+
+			getAppliableColor: function( color ) {
```

Comment:
I thought that would be more coherent with existing `normalizeColor`. But there isn't any real need for that, it can be moved as a private function into `setHtml`.